src/details/ArborX_DetailsCartesianGrid.hpp

```diff
@@ -0,0 +1,109 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILSCARTESIANGRID_HPP
+#define ARBORX_DETAILSCARTESIANGRID_HPP
+
+#include <ArborX_Exception.hpp>
+#include <ArborX_GeometryTraits.hpp>
+
+#include <Kokkos_Macros.hpp>
+
+#include <cassert>
+
+namespace ArborX
+{
+namespace Details
+{
+
+template <class Box>
+struct CartesianGrid
+{
+  static constexpr int DIM = GeometryTraits::dimension<Box>::value;
+  using Point = std::decay_t<decltype(std::declval<Box>().minCorner())>;
+
+  Box _bounds;
+  float _h[DIM];
+  size_t _n[DIM];
```

Comment:
Why are these public?

Reply:
Mostly out of convenience so that we don't have to provide access operators. We print out the sizes in DBSCAN, which is a very useful information.

Reply:
I would prefer a getter function in that case.